libraries/AP_Scripting/lua_bindings.cpp

```diff
@@ -380,9 +380,6 @@ int AP_HAL__I2CDevice_read_registers(lua_State *L) {
     }
 
     AP_HAL::I2CDevice * ud = *check_AP_HAL__I2CDevice(L, 1);
-    if (ud == NULL) {
```

Comment:
can this return nullptr? we deref it below for read_registers and get_semaphore() ? won't this crash

Reply:
No, the check function now checks for null and will not return if the object is null, instead it long jumps to the error handling.